api/src/main/java/me/lokka30/treasury/api/permission/node/holder/NodeHolderPlayer.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * This file is/was part of Treasury. To read more information about Treasury such as its licensing, see <https://github.com/ArcanePlugins/Treasury>.
+ */
+
+package me.lokka30.treasury.api.permission.node.holder;
+
+import java.util.Objects;
+import java.util.Optional;
+import java.util.UUID;
+import java.util.concurrent.CompletableFuture;
+import me.lokka30.treasury.api.common.misc.TriState;
+import me.lokka30.treasury.api.common.response.Response;
+import me.lokka30.treasury.api.permission.context.ContextSet;
+import me.lokka30.treasury.api.permission.node.Node;
+import me.lokka30.treasury.api.permission.node.type.NodeType;
+import org.jetbrains.annotations.NotNull;
+
+public interface NodeHolderPlayer extends NodeHolder {
+
+    @NotNull UUID getUniqueId();
+
+    @Override
+    @NotNull
+    default String getIdentifier() {
+        return this.getUniqueId().toString();
+    }
+
+    /**
+     * Returns whether the specified {@link ContextSet context set} applies for this player node
+     * holder.
+     * <p>By "applies" it is meant that all the
+     * {@link me.lokka30.treasury.api.permission.context.Context contexts} defined in the
+     * specified context set's conditions are met.
+     *
+     * @param contextSet context set to probe
+     * @return applies or not.
+     */
+    boolean contextSetApplies(@NotNull ContextSet contextSet);
+
+    @NotNull
+    default CompletableFuture<Response<TriState>> hasPermission(@NotNull String nodeKey) {
```

Comment:
I'm wondering whether using CompletableFuture here is the best approach. Rather than have a data request be performed every time `hasPermission` is called, why not create a method to pre-load user permission data? The client plugin may request permission data for a user; once loaded, this permission data is in-memory so `hasPermission` calls using the data object return synchronously.  I've never used the LuckPerms API but I believe it allows something like this.

Reply:
I think that should not be in the hands of the API, and rather, the permissions service provider. Regardless, it would not be ideal to add a synchronous cache-assuming `hasPermission` method since that will be abused by developers instead of using the CF version, even when they are not certain that permissions have been cached for a user.

Reply:
What do you think of this idea? We might design the API so that it is impossible to use a cache-backed `hasPermission` without first loading such cache.

For example, a `loadUser(UUID)` method returns a `CompletableFuture<LoadedUser>`. Then, the synchronous `LoadedUser#hasPermission` allows checking permissions synchronously.

Reply:
The API could handle caching, though it is imperative that permission service provider developers fully understand that they must update the cache of a permission holder when some relevant part of their context changes.

For instance, if permission X is only given to player Y in world Z, and player Y changes to world W, then the cache should be updated primarily to ensure that permission X only remains allocated if the player remains in world Z.   

Another consideration to make is that the rest of the APIs do not offer inbuilt caching. Ideally, we keep things consistent between our APIs.

Overall, I'm uncertain about that idea, and more inclined to think it would not be the best route forward.

Reply:
Current design allows for caching. Changing the design to something like `loadUser` will often lead to confusion between both permission plugins and plugins which use the api. 

Consider this: plugin x uses the api and calls `loadUser` each time they need it, which causes multiple database queries to be performed, because developer y of plugin x doesn't know the implementation cache is being updated (if it really is, because permission plugin z may choose not to update it). The current design significantly reduces database queries, because one may not need the whole information about the account.